### PR TITLE
Clipboard Ozone implementation

### PR DESCRIPTION
**Pull request by msisov**
This is a first set of patches, which implement clipboard
support for the Ozone platforms.

The flow is asynchronous and based on Requests. That is,
whenever a ClipboardOzone receives a Read/Write/GetMime
call, it forwards it to the AsyncClipboardOzone, which
then creates a request (used for internal usage and holding
data filled by the ClipboardDelegate), calls to the delegate
and start an abort timer to make sure the request is not stalled.

What is more the clipboard data is cached and removed only when another
chunk of data iss written to a system clipboard. And whenever the chromium browser is the
selection owner, the cached data is used.

### CODE REVIEW

**Review by tonikitoo — COMMENTED** [review]
nitpicking a bit, maybe clipboard_aura.cc or clipboard_auralinux.cc (rather than _linux)?
but lets try to upstream this and hopefully hear from them.

Should this commit include any associated include adaptations?

File: ui/base/BUILD.gn [hunk 1]
```diff
@@ -650,12 +650,21 @@ jumbo_component("base") {
...
     # Aura clipboard.
-    if (use_x11 && is_desktop_linux) {
-      sources += [
-        "clipboard/clipboard_aurax11.cc",
-        "clipboard/clipboard_aurax11.h",
-        "clipboard/clipboard_linux.cc",
-      ]
+    if (is_desktop_linux) {
+      if (use_x11 || use_ozone) {
+        sources += [ "clipboard/clipboard_linux.cc" ]
```
Comment (tonikitoo) [inline ui/base/BUILD.gn:655]: will this file be needed for say ozone/win (something google was upstreaming at some point)?

**Review by tonikitoo — APPROVED** [review]

**Review by tonikitoo — APPROVED** [review]
good thing is that it does not change the ozone's clipboard impl, so I can apply my local patches for middle-click paste as a follow up.

r+

File: ui/base/BUILD.gn (hunk 1 above)
Comment (msisov) [reply ui/base/BUILD.gn:655]: well, this is not platform specific and can be used with mac or win in theory.

File: ui/base/clipboard/clipboard.h [hunk 2]
```diff
@@ -31,6 +31,10 @@
 #include <objidl.h>
 #endif
 
+#if defined(USE_OZONE)
+#include "ui/ozone/public/clipboard_delegate.h"
```
Comment (jkim-julie) [inline ui/base/clipboard/clipboard.h:35]: nit: It would be great if we could use the forward declaration here?

**Comment by jkim-julie** [comment]
lgtm!

File: ui/base/clipboard/clipboard.h (hunk 2 above)
Comment (msisov) [reply ui/base/clipboard/clipboard.h:35]: it will need another set of if defs, which I don't want to add.

**Pull request merged**